File: src/main/java/project/app/flutter_spring_todoapp/todo/domain/TodoPriority.java
package project.app.flutter_spring_todoapp.todo.domain;

import com.fasterxml.jackson.annotation.JsonCreator;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

import java.util.Arrays;
import java.util.Map;
import java.util.stream.Collectors;

@RequiredArgsConstructor
public enum TodoPriority {

    LOW("low"),
    MEDIUM("medium"),
    HIGH("high");

    final String name;
    private final static Map<String,TodoPriority> priorityMap = Arrays.stream(TodoPriority.values())
            .collect(Collectors.toMap(TodoPriority::getName, value -> value));

    @JsonCreator
    public static TodoPriority findPriority(final String source) {
        return priorityMap.get(source.toLowerCase());
    }

    public String getName() {
        return name;
    }

}

File: todo_app/lib/enum/TodoPriority.dart
enum TodoPriority{
  LOW("낮음","low"),
  MEDIUM("중간","medium"),
  HIGH("높음","high");

  final String label;
  final String value;

  const TodoPriority(this.label, this.value);

  // todoPriorityMap을 static으로 선언하고, 초기화
  static final Map<String, TodoPriority> todoPriorityMap =
  Map.fromEntries(TodoPriority.values.map((e) => MapEntry(e.value, e)));

  // String 값을 TodoPriority enum으로 변환하는 메서드
  static TodoPriority fromValue(String value) {
    return todoPriorityMap[value.toLowerCase()] ?? TodoPriority.LOW; // 기본값 LOW
  }
}